presto-orc/src/test/java/com/facebook/presto/orc/TestOrcReaderMemoryUsage.java

```diff
@@ -0,0 +1,295 @@
+        StructField field = objectInspector.getAllStructFieldRefs().get(0);
+
+        for (int i = 0; i < rows; i++) {
+            objectInspector.setStructFieldData(row, field, (long) i);
+            Writable record = serde.serialize(row, objectInspector);
+            writer.write(record);
+        }
+        flushWriter(writer);
+
+        writer.close(false);
+    }
+
+    // Write a file with one column of type BIGINT. There are null values. Other values don't matter.
+    private static void createSingleColumnFileWithNullValues(File file, int rows)
+            throws IOException, ReflectiveOperationException, SerDeException
+    {
+        FileSinkOperator.RecordWriter writer = createOrcRecordWriter(file, ORC_12, CompressionKind.NONE, BIGINT);
+
+        @SuppressWarnings("deprecation") Serializer serde = new OrcSerde();
+        SettableStructObjectInspector objectInspector = createSettableStructObjectInspector("test", BIGINT);
+        Object row = objectInspector.create();
+        StructField field = objectInspector.getAllStructFieldRefs().get(0);
+
+        for (int i = 0; i < rows; i++) {
+            if (i % 10 == 0) {
+                objectInspector.setStructFieldData(row, field, null);
+            }
+            else {
+                objectInspector.setStructFieldData(row, field, (long) i);
+            }
+
+            Writable record = serde.serialize(row, objectInspector);
+            writer.write(record);
+        }
+        flushWriter(writer);
+
+        writer.close(false);
+    }
+
+    // Write a map file with 10 entries in each row. There are null values for each row.
+    private static void createMapFileWithNullValues(File file, Type mapType, int rows)
+            throws IOException, ReflectiveOperationException, SerDeException
+    {
+        FileSinkOperator.RecordWriter writer = createOrcRecordWriter(file, ORC_12, CompressionKind.NONE, mapType);
+
+        Serializer serde = new OrcSerde();
+        SettableStructObjectInspector objectInspector = createSettableStructObjectInspector("test", mapType);
+        Object row = objectInspector.create();
+        StructField field = objectInspector.getAllStructFieldRefs().get(0);
+
+        for (int i = 1; i <= rows; i++) {
+            HashMap<Long, Long> hmap = new HashMap<Long, Long>();
+
+            // Add 10 entries to each row. We added null values so that the nullVector for
+            // the StreamReader is required.
+            for (int j = 1; j <= 8; j++) {
+                Long value = Long.valueOf(j);
+                hmap.put(value, value);
+            }
+            hmap.put(null, Long.valueOf(0));
+            hmap.put(Long.valueOf(0), null);
+
+            objectInspector.setStructFieldData(row, field, hmap);
+            Writable record = serde.serialize(row, objectInspector);
+            writer.write(record);
+        }
+        writer.close(false);
+    }
+
+    private static void flushWriter(FileSinkOperator.RecordWriter writer)
```

Comment:
I feel like we don't need this method (the tests pass without this method, please confirm).

Reply:
yes I copied it from a test writing multi-stripe files. Sorry.